src/Microsoft.OData.Core/JsonLight/ODataJsonLightReader.cs

```diff
@@ -2568,7 +2624,1548 @@ private void ResolveScopeInfoFromContextUrl()
             }
         }
 
-#endregion private methods
+        /// <summary>
+        /// Asynchronous implementation of the reader logic when in state 'Start'.
+        /// </summary>
+        /// <param name="propertyAndAnnotationCollector">The duplicate property names checker to use for the top-level scope.</param>
+        /// <returns>
+        /// A task that represents the asynchronous read operation.
+        /// The value of the TResult parameter contains true if more items can be read from the reader; otherwise false.
+        /// </returns>
+        /// <remarks>
+        /// Pre-Condition:  JsonNodeType.None:      assumes that the JSON reader has not been used yet when not reading a nested payload.
+        /// Post-Condition: when reading a resource set:    the reader is positioned on the first item in the resource set or the end array node of an empty resource set
+        ///                 when reading a resource:  the first node of the first nested resource info value, null for a null expanded link or an end object
+        ///                                         node if there are no navigation links.
+        /// </remarks>
+        private async Task<bool> ReadAtStartImplementationAsynchronously(PropertyAndAnnotationCollector propertyAndAnnotationCollector)

```

Comment:
just use "Async" as suffix?

Reply:
@xuzhg @KenitoInc @habbes So, there's a reason why the `private` methods are suffixed with `Asynchronously` instead of `Async` - _and this goes even for the existing synchronous methods_. It's because the `protected` methods are already using the `Async` suffix. Where you see a `private` `ReadAtStartImplementationAsynchronously` method, there's a `protected` `ReadAtStartImplementationAsync` method. I just followed the pattern I found to avoid introducing confusion.